src/Sentry/Protocol/SentryTraceHeader.cs

```diff
@@ -7,24 +10,60 @@ namespace Sentry.Protocol
     /// </summary>
     public class SentryTraceHeader
     {
-        private readonly SentryId _traceId;
-        private readonly SpanId _spanId;
-        private readonly bool? _isSampled;
+        private const string HeaderName = "sentry-trace";
+
+        /// <summary>
+        /// Trace ID.
+        /// </summary>
+        public SentryId TraceId { get; }
+
+        /// <summary>
+        /// Span ID.
+        /// </summary>
+        public SpanId SpanId { get; }
+
+        /// <summary>
+        /// Whether the trace is sampled.
+        /// </summary>
+        public bool? IsSampled { get; }
 
         /// <summary>
         /// Initializes an instance of <see cref="SentryTraceHeader"/>.
         /// </summary>
-        public SentryTraceHeader(SentryId traceId, SpanId spanId, bool? isSampled)
+        public SentryTraceHeader(SentryId traceId, SpanId spanSpanId, bool? isSampled)
         {
-            _traceId = traceId;
-            _spanId = spanId;
-            _isSampled = isSampled;
+            TraceId = traceId;
+            SpanId = spanSpanId;
+            IsSampled = isSampled;
         }
 
+        /// <summary>
+        /// Injects trace information into HTTP headers.
+        /// </summary>
+        public void Inject(HttpHeaders headers)
+        {
+            var headerValue = ToString();
+
+            headers.Remove(HeaderName);
+            headers.Add(HeaderName, headerValue);
+        }
+
+        /// <summary>
+        /// Injects trace information into the headers of the specified HTTP request.
+        /// </summary>
+        public void Inject(HttpRequestMessage request) =>
+            Inject(request.Headers);
+
+        /// <summary>
+        /// Injects trace information into the default headers of the specified HTTP client.
+        /// </summary>
+        public void Inject(HttpClient client) =>
+            Inject(client.DefaultRequestHeaders);
```

Comment:
This facilitates this type of workflow:

```csharp
using var client = new HttpClient();
using var request = new HttpRequestMessage(HttpMethod.Get, "https://example.com");
SentrySdk.GetTraceHeader().Inject(request);
```

Could make it as extension methods instead on `HttpHeaders`/`HttpRequestMessage`/`HttpClient` instead but not sure if worth.

Reply:
I'm hoping we'll be able to run this through an integration instead of users having to do this manually on their `HttpClient`s. Maybe as a `MessageHandler` or even through ListenerHandler if that's possible:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/2bc7924b065ddef8199bf69c2456724980522557/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

https://github.com/open-telemetry/opentelemetry-dotnet/blob/2bc7924b065ddef8199bf69c2456724980522557/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpInstrumentationEventSource.cs

Reply:
These would also be useful from inside the handlers, I believe. I guess I could also add another method `GetHttpHandler()` that wraps an existing handler to make it easier for those use cases. I'll look into `ListenerHandler`, first time hearing about it.